v-next/hardhat/src/internal/cli/telemetry/sentry/anonymizer.ts

```diff
@@ -166,38 +215,47 @@ export class Anonymizer {
         continue;
       }
 
-      if (this.#errorRaisedByPackageToIgnore(frame.filename)) {
+      // We don't report errors from the Hardhat.config file
+      if (
+        this.#configPath !== undefined &&
+        this.#configPath.includes(frame.filename)
+      ) {
         return false;
       }
 
-      // we stop after finding either a hardhat file or a file from the user's
-      // project
-      if (this.#isHardhatFile(frame.filename)) {
-        return true;
-      }
-
-      if (frame.filename === ANONYMIZED_FILE) {
+      // We don't report errors from the ignored package list e.g. `ethers`
+      if (this.#errorRaisedByPackageToIgnore(frame.filename)) {
         return false;
       }
 
+      // We don't report errors originating from non-Hardhat packages
       if (
-        this.#configPath !== undefined &&
-        this.#configPath.includes(frame.filename)
+        this.#isPackageFile(frame.filename) &&
+        !this.#isHardhatFile(frame.filename)
       ) {
         return false;
       }
```

Comment:
There is an argument to say this is too aggresive at the first pass, and we should disable this heuristic for the moment.